sips/pending/_posts/2017-01-13-binary-compatibility.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: sip
+title: SIP XX - Improving binary compatibility with @binaryCompatible
+disqus: true
+---
+
+__Dmitry Petrashko__
+
+__first submitted 13 February 2017__
+
+## Introduction ##
+
+Scala is a language which evolves fast and thus made a decision to only promise binary compatibility across minor releases.
+At the same time, there is a demand to develop APIs that live longer than a major release cycle of Scala.
+This SIP introduces an annotation `@binaryCompatible` that checks that `what you write is what you get`.
+It will fail compilation in case emitted methods or their signatures 
+are different from those written by users. 
+As long as signatures of methods in source is not changed, `@binaryCompatible` annotated class 
+will be compatible across major version of Scala. 
+
+## Use Cases
+Dotty currently uses java defined interfaces as public API for SBT in order to ensure binary compatibility. 
```

Comment:
Actually `dotty-interfaces` is not used by sbt at all. `dotty-sbt-bridge` depends on dotty itself, and sbt will automatically compile it from source against the current version of `dotty` (this is nothing new, the same thing happens with the scalac sbt bridge).
However, `dotty-interfaces` is used by IntelliJ IDEA.